src/IECoreMaya/ToMayaPlugConverter.cpp

```diff
@@ -68,12 +71,100 @@ ToMayaPlugConverterPtr ToMayaPlugConverter::create( const IECore::ObjectPtr src
 	return new ToMayaPlugConverter( src );
 }
 
-bool ToMayaPlugConverter::convert( MPlug &plug ) const
+template <typename DataType, typename UnitType>
+MStatus ToMayaPlugConverter::setPlugValue(MPlug &plug, boost::intrusive_ptr<const IECore::TypedData<DataType> > data)
 {
-	ConstObjectPtr toConvert = srcParameter()->getValidatedValue();
+	MStatus s;
+	s = plug.setValue(data->readable());
+	return s;
+}
+
 
+// specializations for Unit Attributes
+namespace IECoreMaya
+{
+
+template <>
+MStatus ToMayaPlugConverter::setPlugValue<float, MTime>(MPlug &plug, boost::intrusive_ptr<const IECore::TypedData<float> > data)
+{
 	MStatus s;
+	MTime t;
+	t.setUnit(MTime::kSeconds);
+	t.setValue(data->readable());
+	s = plug.setValue(t);
+	return s;
+}
+
+template <>
+MStatus ToMayaPlugConverter::setPlugValue<float, MDistance>(MPlug &plug, boost::intrusive_ptr<const IECore::TypedData<float> > data)
+{
+	MStatus s;
+	MDistance t;
+	t.setUnit(MDistance::kCentimeters);
+	t.setValue(data->readable());
+	s = plug.setValue(t);
+	return s;
+}
+
+template <>
+MStatus ToMayaPlugConverter::setPlugValue<float, MAngle>(MPlug &plug, boost::intrusive_ptr<const IECore::TypedData<float> > data)
+{
+	MStatus s;
+	MAngle t;
+	t.setUnit(MAngle::kRadians);
+	t.setValue(data->readable());
+	s = plug.setValue(t);
+	return s;
+}
+
+} // namespace IECoreMaya
+
+
+template <typename DataType, typename UnitType>
+MStatus ToMayaPlugConverter::convertAttr(MPlug &plug) const
+{
+	MStatus s;
+	if (plug.isArray())
+	{
+		ConstObjectPtr toConvert = srcParameter()->getValidatedValue();
+		IECore::CompoundData::ConstPtr data = IECore::runTimeCast<const IECore::CompoundData>( toConvert );
 
+		typedef TypedData< std::vector<DataType> > VecDataType;
+		const VecDataType *indexedData = data->member<VecDataType>( "data", false );
+		const IntVectorData *indices = data->member<IntVectorData>( "indices", false );
+
+		// set all indexed values
+		assert( indexedData->readable().size() == indices->readable().size() );
+		for (int i=0; i<indexedData->readable().size(); ++i)
+		{
+			boost::intrusive_ptr<IECore::TypedData<DataType> > p = new IECore::TypedData<DataType>;
+			p->writable() = indexedData->readable().at(i);
+			MPlug currentPlug = plug.elementByLogicalIndex(indices->readable().at(i));
+			if (data == 0)
+			{
+				return s;
+			}
+			s = setPlugValue<DataType, UnitType>(currentPlug, p);
```

Comment:
we're stomping over the return status as we loop. if we're not going to check it each time, no need to record it really.

Reply:
oh yeah that should throw in case the status isn't kSuccess

Reply:
yep that should return in case of a failure